packages/simple-react-editors/src/SimpleReactEditor.tsx

```diff
@@ -41,36 +39,42 @@ export class SimpleReactEditor extends React.Component<Props, State> {
     this.props.messageBus.notify_ready();
   }
 
-  public setContent(content: string) {
-    return new Promise<void>(res => this.setState({ originalContent: content }, res)).then(() =>
-      this.updateContent(content)
-    );
+  public async setContent(content: string): Promise<void> {
+    this.setState({ content: content });
+  }
+
+  private async updateContent(content: string): Promise<void> {
+    // The updateContent method is also called when users perform undo/redo actions
+    // but, ideally, messageBus.notify_newEdit shouldn't be called in this cases.
+    this.props.messageBus.notify_newEdit({ id: new Date().getTime().toString() });
+    this.setContent(content);
   }
 
-  private updateContent(content: string) {
-    return new Promise<void>(res => {
-      this.setState({ content: content }, () => {
-        this.props.messageBus.notify_dirtyIndicatorChange(this.isDirty());
-        res();
-      });
-    });
+  public async getContent(): Promise<string> {
+    return this.state.content;
   }
 
-  //saving triggers this method, so we also update the originalContent by calling `this.setContent`
-  public getContent() {
-    return this.setContent(this.state.content).then(() => this.state.content);
+  public isDirty(): boolean {
+    throw new Error("Method not implemented.");
   }
```

Comment:
I think we should deprecate this method on `kogito-tooling`. This is not used and we have a different way of identifying that a file has been edited.

Reply:
+1